### PR TITLE
issue 320 - added support for specifying surefire/TestNG groups at ru…

### PR DESCRIPTION
**Pull request by iancorcoran**
…ntime. also added support for passing api key details via system properties

### CODE REVIEW

File: src/main/groovy/com/stormpath/tck/util/ApiKey.groovy [hunk 1]
```diff
@@ -60,13 +60,21 @@ class ApiKey {
            apiKeySecret = properties.get(API_KEY_SECRET_PROPERTY_NAME) != null ? properties.get(API_KEY_SECRET_PROPERTY_NAME) : apiKeySecret
        }
 
+       //4. Try system properties:
+       properties = getSystemProperties();
+       if (properties.size() > 0) {
+           apiKeyID = properties.get(API_KEY_ID_PROPERTY_NAME) != null ? properties.get(API_KEY_ID_PROPERTY_NAME) : apiKeyID
```
Comment (nbarbettini) [inline src/main/groovy/com/stormpath/tck/util/ApiKey.groovy]: These properties are deprecated - is Java still using them?

Comment (nbarbettini) [inline src/main/groovy/com/stormpath/tck/util/ApiKey.groovy:66]: These property names are deprecated - is Java still using them?

**Review by nbarbettini — CHANGES_REQUESTED** [review]
Looks good, other than the deprecated env var names.

File: src/main/groovy/com/stormpath/tck/util/ApiKey.groovy (hunk 1 above)
Comment (iancorcoran) [reply src/main/groovy/com/stormpath/tck/util/ApiKey.groovy:66]: The Java SDK does support reading in ApiKey credentials via system properties, but this is separate. It is just a convenience mechanism to read in the creds the TCK uses to communicate with IAM and not part of the testing the integration, so not sure I quite understand the objection?

Comment (nbarbettini) [reply src/main/groovy/com/stormpath/tck/util/ApiKey.groovy:66]: Ah, I understand. Nevermind, I retract my question. :)

**Review by nbarbettini — APPROVED** [review]

**Comment by nbarbettini** [comment]
LGTM 👍

**Pull request merged**